src/shogun/classifier/Perceptron.h

```diff
@@ -72,28 +60,21 @@ class CPerceptron : public CLinearMachine
 
 	protected:
 		/** registers and initializes parameters */
-			void init();
-
-		/** train classifier
-		 *
-		 * @param data training data (parameter can be avoided if distance or
-		 * kernel-based classifiers are used and distance/kernels are
-		 * initialized with train data)
-		 *
-		 * @return whether training was successful
-		 */
-		virtual bool train_machine(CFeatures* data=NULL);
+		void init();
+		/** init the model before training begins */
+		virtual void init_model(CFeatures* data);
+		/** perform a single iteration over features and update weights */
```

Comment:
same here, we would only overload the docs if something special happens inside the method (which it doesnt for perceptron)